src/interfaces/vault.spec.ts

```diff
@@ -0,0 +1,868 @@
+          [tokenMock.address]: "token-mock-icon.png"
+        });
+        helperTokensMock.mockReturnValue(
+          Promise.resolve<Token[]>([tokenMock])
+        );
+        tokensMetadataMock.mockReturnValue(
+          Promise.resolve<TokenMetadata[]>([{ ...tokenMock, description: "Token mock metadata" }])
+        );
+        oracleGetPriceUsdcMock.mockResolvedValue("1");
+        const fillTokenMetadataOverridesMock = jest.fn();
+        (vaultInterface as any).fillTokenMetadataOverrides = fillTokenMetadataOverridesMock;
+
+        const actualTokens = await vaultInterface.tokens();
+
+        expect(actualTokens).toEqual([
+          {
+            ...tokenMock,
+            icon: "token-mock-icon.png",
+            metadata: {
+              address: "0x001",
+              decimals: "18",
+              description: "Token mock metadata",
+              symbol: "DEAD",
+              name: "Dead Token",
+              priceUsdc: "0",
+              supported: {
+                zapper: true
+              }
+            },
+            name: "Dead Token",
+            priceUsdc: "1",
+            supported: {}
+          }
+        ]);
+        expect(lensAdaptersVaultsV2TokensMock).toHaveBeenCalledTimes(1);
+        expect(lensAdaptersVaultsV2TokensMock).toHaveBeenCalledWith(undefined); // no overrides
+        expect(assetIconMock).toHaveBeenCalledTimes(1);
+        expect(assetIconMock).toHaveBeenCalledWith(["0x001", EthAddress]);
+        expect(helperTokensMock).toHaveBeenCalledTimes(1);
+        expect(helperTokensMock).toHaveBeenCalledWith(["0x001"], undefined);
+        expect(tokensMetadataMock).toHaveBeenCalledTimes(1);
+        expect(tokensMetadataMock).toHaveBeenCalledWith(["0x001"]);
+        expect(assetAliasMock).toHaveBeenCalledTimes(1);
+        expect(assetAliasMock).toHaveBeenCalledWith("0x001");
+        expect(fillTokenMetadataOverridesMock).toHaveBeenCalledTimes(1);
+        expect(fillTokenMetadataOverridesMock).toHaveBeenCalledWith(actualTokens[0], actualTokens[0].metadata);
+      });
+    });
+  });
+
+  describe("deposit", () => {
+    describe("when is zapping into pickle jar", () => {
+      beforeEach(() => {
+        PickleJarsMock.PickleJars = ["0xVault"];
+      });
+
+      it("should call zapIn with correct arguments and pickle as the zapProtocol", async () => {
+        const zapInMock = jest.fn();
+        (vaultInterface as any).zapIn = zapInMock;
+
+        const [vault, token, amount, account] = ["0xVault", "0xToken", "1", "0xAccount"];
+
+        await vaultInterface.deposit(vault, token, amount, account);
+
+        expect(zapInMock).toHaveBeenCalledTimes(1);
+        expect(zapInMock).toHaveBeenCalledWith(
+          vault,
+          token,
+          amount,
+          account,
+          { sendTransaction: sendTransactionMock },
+          {},
+          ZapProtocol.PICKLE,
+          {}
+        );
+      });
+    });
+
+    describe("when is not zapping into pickle jar", () => {
```

Comment:
so what is it calling when it is not a pickle jar?

Reply:
It depends, the describes inside this one give you the options

```typescript
describe("when vault ref token is the same as the token", () => {
  describe("when token is eth address", () => {
  });

  describe("when token is not eth address", () => {
    it("should deposit into a yearn vault", async () => {
    });
  });
});

describe("when vault ref token is not the same as the token", () => {
  it("should call zapIn with correct arguments and yearn as the zapProtocol", async () => {
});
```